AD74413R updates

CiprianRegus opened the pull request:
## Pull Request Description

Fixes/ new features related to the AD74413R driver. Required for the SWIOT1L project.

## PR Type
- [x] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -1085,6 +1085,24 @@ int ad74413r_remove(struct ad74413r_desc *desc)
 		return -EINVAL;
 
+	/* Perform a reset to bring the device in the default state */
+	if (desc->reset_gpio) {
```
this looks like duplicated code. A separate function for hardware reset might fit. And then call it both in init function and remove function.

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -156,7 +156,7 @@ int ad74413r_dac_voltage_to_code(uint32_t mvolts, uint32_t *code)
  * @param val - Register value.
  * @param buff - The communication buffer.
  */
-static void ad74413r_format_reg_write(uint8_t reg, uint16_t val, uint8_t *buff)
```
the data width for this device is exactly 16-bit, according to datasheet, therefore we already reduce the risk considerably of sending random data that might be passed with 32-bit wide values when not properly initialized. Moreover, is it a good idea to adapt a driver based on the upper layers? It should be vice-versa from my point of view.

CiprianRegus replied:
for this exact cast there is no risk of sending random data, since the `no_os_put_unaligned_be16()` takes a `uint16_t` as a parameter, thus it will cast it to 16 bit (meaning that the upper 16 bits will just be ignored). However I agree that since this function doesn't receive a pointer to the `val` parameter there is no good reason for this change and I can revert it. The behavior is the same in both cases.

What I meant by "requiring extra handling in the upper layers" is the following case: `ad74413r_get_adc_range(struct ad74413r_desc *, uint32_t, uint16_t *)` -  this function has to receive a pointer to a 16 bit wide integer. Passing a pointer to `uint32_t` on a little endian cpu will result in writing at the wrong address. I think it would be a better API to just use pointers to 32 bit integers for all the cases where a value has to be read.

buha replied:
I think this 16 -> 32 bits change may introduce bugs

For example line 162 here, i think it might not work on any endianess because it's now picking 2 bytes out of 4 and rearranges them. With val being 16 bits, it would work as intended, picking 2 bytes out of 2 (all good!) and rearranging them... 
https://github.com/analogdevicesinc/no-OS/blob/a78abcb369f233b9086c0ccf8ae0287e761f4648/drivers/adc-dac/ad74413r/ad74413r.c#L159-L164

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -312,8 +311,8 @@ int ad74413r_set_info(struct ad74413r_desc *desc, uint16_t mode)
 	int ret;
-	uint16_t val;
-	uint16_t test_val = 0x1234;
+	uint32_t val = 0;
```
is this explicit initialization required?

CiprianRegus replied:
it was added by mistake, will remove.

buha commented on drivers/adc-dac/ad74413r/iio_ad74413r_trig.c:
```diff
@@ -47,7 +47,7 @@
 /******************************************************************************/
 struct iio_trigger ad74413r_iio_trig_desc = {
-	.is_synchronous = false,
+	.is_synchronous = true,
```
This seems like project specific behavior and makes me wonder why iio_ad74413r_trig.c file is where it is.
It seems the ad74413r_iio_trig_desc definition should be in application code and not under drivers/.

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -364,6 +374,17 @@ int ad74413r_set_channel_function(struct ad74413r_desc *desc,
+		return ret;
+
+	/* Each function should be selected for at least 130 us. */
+	no_os_udelay(150);
```
nit: it should be 130 here.

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -338,13 +337,24 @@ int ad74413r_reset(struct ad74413r_desc *desc)
-		return ret;
+		/* Minimum RESET signal pulse duration */
+		no_os_udelay(50);
+		ret = no_os_gpio_set_value(desc->reset_gpio, NO_OS_GPIO_HIGH);
```
missing error check. It will return 0 even though gpio_set_value fails.

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -338,13 +337,24 @@ int ad74413r_reset(struct ad74413r_desc *desc)
-	ret = ad74413r_reg_write(desc, AD74413R_CMD_KEY, AD74413R_CMD_KEY_RESET_1);
-	if (ret)
-		return ret;
+	if (desc->reset_gpio) {
```
please update the function documentation since now it does not do only a "soft reset".

CiprianRegus commented:
- Removed the commit which changed the function parameters to be `uint32_t`, instead fixed incorrect parameter passing in some places.
- Removed the `iio_ad74413r_trig.c` file, since the trigger type should be specific to each project.
- Addressed all the other review comments

rbolboac commented on drivers/adc-dac/ad74413r/iio_ad74413r.c:
```diff
@@ -309,6 +310,8 @@ static int ad74413r_iio_read_offset(void *dev, char *buf, uint32_t len,
 			ad74413r_get_adc_range(iio_desc->ad74413r_desc,
 					       channel->address, &reg_val);
+			if (ret)
+				return ret;
```
am I missing something here? I don't see where ret value is set

CiprianRegus replied:
`ret` has to be set, will change.

buha reviewed (approved):
minus that remark from Ramona, looks really good
